Python/import.c

```diff
@@ -1683,7 +1740,13 @@ update_global_state_for_extension(PyThreadState *tstate,
 #endif
         cached = _extensions_cache_set(
                 path, name, def, m_init, singlephase->m_index, m_dict,
-                singlephase->origin);
+                singlephase->origin,
+#ifdef Py_GIL_DISABLED
+                singlephase->md_gil
+#else
+                NULL
+#endif
```

Comment:
This is pretty awkward. It could be avoided by doing `cached->md_gil = singlephase->md_gil` right after the `_extensions_cache_set()` call, in this function.

The main downside of that is that it's less consistent with the other data transferred from `singlephase` to `cached`, which all happens inside `_extensions_cache_set()`. I'm happy to do this another way if anyone feels strongly.

Reply:
Yeah, I'd be fine with always adding md_gil.